cypress/integration/teacher_add_new_unregistered_student.spec.js

```diff
@@ -0,0 +1,189 @@
+import studentHomePageElements from '../support/elements/student_home_page_elements.js'
+import registerAddStudentPageElements from '../support/elements/register_add_student_page_elements.js'
+import adminPageElements from '../support/elements/admin_page_elements.js'
+import adminSettingsUsersPageElements from '../support/elements/admin_settings_users_page_elements.js'
+
+const STUDENT_TWO_NAME = constants.STUDENT_TWO_LASTNAME + ", " + constants.STUDENT_TWO_FIRSTNAME;
+const STUDENT_TWO_FULLNAME = constants.STUDENT_TWO_FIRSTNAME + " " + constants.STUDENT_TWO_LASTNAME;
+const STUDENT_TWO_USERNAME = constants.STUDENT_TWO_USERNAME;
+const STUDENT_TWO_PASSWORD = constants.STUDENT_TWO_PASSWORD;
+const STUDENT_THREE_NAME = constants.STUDENT_THREE_LASTNAME + ", " + constants.STUDENT_THREE_FIRSTNAME;
+const STUDENT_THREE_FULLNAME = constants.STUDENT_THREE_FIRSTNAME + " " + constants.STUDENT_THREE_LASTNAME;
+const STUDENT_THREE_USERNAME = constants.STUDENT_THREE_USERNAME;
+const STUDENT_THREE_PASSWORD = constants.STUDENT_THREE_PASSWORD;
+const STUDENT_NEW_PASSWORD = STUDENT_TWO_PASSWORD + "1";
+let className = undefined;
+
+context("Verify teacher can add a new student to a class", () => {
+
+  before(function() {
+    cy.visit(constants.LEARN_PORTAL_BASE_URL); // Visit LEARN Portal home page
+    cy.login(constants.TEACHER_USERNAME, constants.TEACHER_PASSWORD); // Login as teacher user
+  });
+
+  after(function() {
+    cy.logout();
+  });
+
+  it("Verify teacher is able to unarchive a class", () => {
+    cy.contains(teacherHomePageElements.LEFT_NAV_CLASSES.selector, "Classes").click(); // Click 'Classes' Section
+    cy.get(teacherHomePageElements.LEFT_NAV_MANAGE_CLASSES.selector).click(); // Click 'Manage Classes' Section
+    cy.contains(manageClassesPageElements.LAST_CLASS_ARCHIVE_UNARCHIVE.selector, "Unarchive").click(); // Unarchive the last class in the list
+    cy.get(manageClassesPageElements.LAST_CLASS_NAME.selector).then(($lastclass) => {
+      className = $lastclass.text(); // Get the class name of the unarchived class
+    });
+    cy.get(userHomePageElements.HEADER_MYCLASSES.selector).click(); // Click 'My Classes' in the top header
+    cy.contains(teacherHomePageElements.LEFT_NAV_CLASSES.selector, "Classes").click(); // Expand 'Classes' in the left nav
+    cy.get(teacherHomePageElements.LEFT_NAV_CLASS_NAME.selector).click(); // Expand unarchived class in the left nav
+  });
+
+  it("Verify teacher is able to add new students to the class", () => {
+    cy.contains(teacherHomePageElements.LEFT_NAV_CLASS_NAME.selector, className).within(() => {
+      cy.get(teacherHomePageElements.LEFT_NAV_STUDENT_ROSTER.selector).click(); // Click 'Student Roster' for the current class
+    });
+    cy.get(studentRosterPageElements.CLASS_COUNT.selector).should("have.text", "0"); // Check 'Student Count' is '0'
+    cy.get(studentRosterPageElements.REGISTER_ADD_STUDENT_LINK.selector).click(); // Click 'Register & Add Student' link
```

Comment:
It would be useful to make a list of all the places where new objects or records are being created by the tests. Perhaps this could just be through comments with a certain format. 
This way we can see how much the tests might be growing the database each time they are run, and also we can track which ones there is a way to delete and which ones there isn't a way to delete.

Reply:
I've added a comment in every file to track this. Below are a couple examples from 2 different spec files:

- // Note for db tracking : No db tracking required, using existing records (except for notices, which are added and then deleted at the end)
- // Note for db tracking : This test adds a class at the start and then archives it at the end

For the current test coverage, the only place I am adding something and not deleting at the end of the tests is creating a new class (I didn't see a way to do it from the UI, so I archive it instead. But if you find a way to delete a class, then I can implement. But for all the remaining tests, whatever records I create - students, notices, etc I am deleting them at the end.